### ISSUE TITLE
r2 -d empty_file

### ISSUE THREAD

**Issue by zonkzonk** [radare/radare2#2846]
morrn,

Update:
chmod +x empty_file
r2 -qc "aa;pdr;"  -d empty_file

```
zlul@debian:~/crashes/crashes.fuzzing-project.org$ gdb -q r2 core 
Reading symbols from /usr/local/bin/r2...done.
[New LWP 16174]

warning: Can't read pathname for load map: Input/output error.
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/i386-linux-gnu/i686/cmov/libthread_db.so.1".
Core was generated by `r2 -qc aa;pdr; -d %1269168516chmlib-floating-point-exception.chm'.
Program terminated with signal 11, Segmentation fault.
#0  0xb752a4c3 in baddr (arch=0x9adc328)
    at /home/zlul/radare2/libr/..//libr/bin/p/bin_art.c:77
77              return ao->art.image_base;
(gdb) bt
#0  0xb752a4c3 in baddr (arch=0x9adc328)
    at /home/zlul/radare2/libr/..//libr/bin/p/bin_art.c:77
#1  0xb751918f in r_bin_object_set_items (binfile=0x9adc328, o=0x9ae3ae8)
    at bin.c:408
#2  0xb751ae80 in r_bin_object_new (binfile=0x9adc328, plugin=0x9aa0fd0, 
    baseaddr=134512640, loadaddr=134512640, offset=0, sz=0) at bin.c:945
#3  0xb751b51d in r_bin_file_new_from_bytes (bin=0x9aa0de0, 
    file=0x9adc4a8 "./%1269168516chmlib-floating-point-exception.chm", 
    bytes=0x9adc6b0 "", sz=0, file_sz=0, rawstr=0, baseaddr=134512640, 
    loadaddr=134512640, fd=16175, pluginname=0x0, xtrname=0x0, offset=0)
    at bin.c:1056
#4  0xb751a0c4 in r_bin_load_io_at_offset_as_sz (bin=0x9aa0de0, 
    desc=0x9ad14b8, baseaddr=134512640, loadaddr=134512640, xtr_idx=0, 
    offset=0, name=0x0, sz=0) at bin.c:644
#5  0xb751a1c6 in r_bin_load_io_at_offset_as (bin=0x9aa0de0, desc=0x9ad14b8, 
    baseaddr=134512640, loadaddr=134512640, xtr_idx=0, offset=0, name=0x0)
    at bin.c:666
#6  0xb7519aec in r_bin_load_io (bin=0x9aa0de0, desc=0x9ad14b8, 
    baseaddr=134512640, loadaddr=134512640, xtr_idx=0) at bin.c:546
#7  0xb7519527 in r_bin_load (bin=0x9aa0de0, 
    file=0x9ad2b2e "./%1269168516chmlib-floating-point-exception.chm", 
    baseaddr=134512640, loadaddr=134512640, xtr_idx=0, fd=16175, rawstr=0)
    at bin.c:452
---Type <return> to continue, or q <return> to quit---
```

Greetings
--zlul

**Comment by zonkzonk** [comment]
r2 -v:
radare2 0.10.0-git 8247 @ linux-little-x86-32 git.0.9.9-148-gd5f2661
commit: d5f2661cbe1a32bc26490bd7a1864ef45907aaea build: 2015-06-27

**Comment by crowell** [comment]
cant repro

**Comment by zonkzonk** [comment]
```
zlul@debian:~/crashes/crashes.fuzzing-project.org$ for x in *; do r2 -qc "aa;pdr;"  -d $x; if test -f core; then echo "$x"; read; fi; done
Process with PID 3829 started...
PID = 3829
pid = 3829 tid = 3829
r_debug_select: 3829 3829
Using BADDR 0x8048000
Asuming filepath ./%1269168516chmlib-floating-point-exception.chm
Segmentation fault
Process with PID 3835 started...
PID = 3835
pid = 3835 tid = 3835
r_debug_select: 3835 3835
Using BADDR 0x8048000
Asuming filepath ./%1269168516chmlib-floating-point-exception.chm
Segmentation fault
Process with PID 3841 started...
PID = 3841
pid = 3841 tid = 3841
r_debug_select: 3841 3841
Using BADDR 0x8048000
Asuming filepath ./chmlib-floating-point-exception.chm
bits 32
pid = 3841 tid = 3841
Interrupted
This block size is too big (52428800<-5227385866824974262). Did you mean 'pD @ 0xb77498400000004a' instead?
```
and so on. 32bit Debian 7.

**Comment by zonkzonk** [comment]
@crowell I can provide core file in private

**Comment by zonkzonk** [comment]
@all r2 -d empty_file -> segfaults here

**Comment by crowell** [comment]
```
minishwoods tools/radare2 ‹master*› » touch mt               
minishwoods tools/radare2 ‹master*› » r2 -qc "aa;pdr;" -d mt 
p/io_debug.c:334 fork_and_attach: execv: Permission denied   
PID = -1                                                     
r_debug_native_map_get: No selected pid (-1)                 
Cannot open 'dbg://./mt' for writing.                        
```

still cant repro

**Comment by alvarofe** [comment]
Me neither

```
alvaro_fe@macbook  ~/Projects/Reverse/radare2/r2   master ± sudo r2 -qc "aa;pdr;" -d empty_file
Process with PID 23732 started...
PID = 23732
pid: 23732
task: 2563
pid = 23732 tid = 23732
r_debug_select: 23732 23732
Using BADDR 0x102a70000
bits 32
pid = 23732 tid = 23732
Try: '=!pid'
debug_getregs: Failed to get thread 23732 2563.error (4). ((os/kern) invalid argument)
p/debug_native.c:1189 thread_get_state: Invalid argument
debug_getregs: Failed to get thread 23732 2563.error (4). ((os/kern) invalid argument)
p/debug_native.c:1189 thread_get_state: Invalid argument
Cannot find function at 0x00000000
```

**Comment by zonkzonk** [comment]
@crowell  I had chmod +x empty file

**Comment by alvarofe** [comment]
In debian crash 

```
alvaro@warzone 16:25 ~/Projects/Reverse/radare2/r2 r2 -qc "add;pdr;" -d empty_file                                                                               [master] git:r2
Process with PID 15012 started...
PID = 15012
pid = 15012 tid = 15012
r_debug_select: 15012 15012
Using BADDR 0x555555554000
Asuming filepath ./empty_file
ASAN:SIGSEGV
=================================================================
==15011==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000010 (pc 0x7ffff6310af3 bp 0x7fffffffa010 sp 0x7fffffff9f00 T0)
    #0 0x7ffff6310af2 in baddr /home/alvaro/Projects/Reverse/radare2/r2/libr/..//libr/bin/p/bin_art.c:77:2
    #1 0x7ffff628db9b in r_bin_object_set_items /home/alvaro/Projects/Reverse/radare2/r2/libr/bin/bin.c:408:28
    #2 0x7ffff627c6e3 in r_bin_object_new /home/alvaro/Projects/Reverse/radare2/r2/libr/bin/bin.c:945:2
    #3 0x7ffff626c508 in r_bin_file_new_from_bytes /home/alvaro/Projects/Reverse/radare2/r2/libr/bin/bin.c:1056:6
    #4 0x7ffff62690d6 in r_bin_load_io_at_offset_as_sz /home/alvaro/Projects/Reverse/radare2/r2/libr/bin/bin.c:644:13
    #5 0x7ffff6262285 in r_bin_load_io_at_offset_as /home/alvaro/Projects/Reverse/radare2/r2/libr/bin/bin.c:666:12
    #6 0x7ffff6260dc3 in r_bin_load_io /home/alvaro/Projects/Reverse/radare2/r2/libr/bin/bin.c:546:9
    #7 0x7ffff62608ba in r_bin_load /home/alvaro/Projects/Reverse/radare2/r2/libr/bin/bin.c:452:9
    #8 0x7ffff768d7fd in r_core_file_do_load_for_debug /home/alvaro/Projects/Reverse/radare2/r2/libr/core/file.c:307:7
    #9 0x7ffff7689ddd in r_core_bin_load /home/alvaro/Projects/Reverse/radare2/r2/libr/core/file.c:485:4
    #10 0x4c5283 in main /home/alvaro/Projects/Reverse/radare2/r2/binr/radare2/radare2.c:544:10
    #11 0x7ffff00a6b44 in __libc_start_main /build/glibc-Ir_s5K/glibc-2.19/csu/libc-start.c:287
    #12 0x4bf1fc in _start (/home/alvaro/Projects/Reverse/radare2/r2/binr/radare2/radare2+0x4bf1fc)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV /home/alvaro/Projects/Reverse/radare2/r2/libr/..//libr/bin/p/bin_art.c:77 baddr
==15011==ABORTING

```

**Comment by zonkzonk** [comment]
ah :)

**Comment by zonkzonk** [comment]
confirmed fix in be0c7cf457f8578fe26f4ebb42eb64ab0c45e0d0
